common/src/main/java/feast/common/function/StringUtils.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.function;
+
+import feast.proto.core.FeatureSetProto.FeatureSetSpec;
+import feast.proto.core.FeatureSetReferenceProto.FeatureSetReference;
+import feast.proto.core.StoreProto;
+import feast.proto.serving.ServingAPIProto.FeatureReference;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class StringUtils {
```

Comment:
I would rather have those functions grouped by domain. Also name "StringUtils" doesn't bring clarity here

Reply:
I was expecting this haha, had a hard time thinking for a name so pushed for a review to have you guys help brainstorm 😂

Reply:
By domain you mean `FeatureSet`, `Store` etc?

Reply:
My trigger words are
* Info
* Spec
* Util
* Helper
* Metadata

Sometimes unavoidable, but we should pay attention when/why they are introduced.

Reply:
> By domain you mean `FeatureSet`, `Store` etc?

exactly

Reply:
Updated classes based on domain.